core/src/main/java/com/linecorp/armeria/unsafe/ByteBufHttpData.java

```diff
@@ -40,13 +41,20 @@
     private final int length;
 
     /**
-     * Construct a new {@link ByteBufHttpData}. Ownership of {@code buf} is taken by this
+     * Constructs a new {@link ByteBufHttpData}. Ownership of {@code buf} is taken by this
      * {@link ByteBufHttpData}, which must not be mutated anymore.
      */
     public ByteBufHttpData(ByteBuf buf, boolean endOfStream) {
-        this.buf = buf;
+        final int length = requireNonNull(buf, "buf").readableBytes();
```

Comment:
Don't see a big reason to not just use the field without the local variable.

Reply:
Like this?

    length = buf.readableBytes();
    if (length != 0) { ... }

Reply:
Yup - the field is final too, so seems to be the same

Reply:
Fixed